### PR TITLE
[doc] Update documentation for github and gitlab backends

### PR DESCRIPTION
**Pull request by vchrombie**
This code updates the information about the categories of the github and gitlab backends in the README and bin/perceval. Furthermore, it improves the docstrings of the methods `pulls` and `issues` of the GitHubClient.

Please check and suggest me changes if you need any. I would be happy to make the changes in this same PR.
@jgbarah @valeriocos 

Sorry for the delay in PR as I was busy with some exams in the last week.
Solves #494 if this pr is merged.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,8 +18,8 @@ are:
     dockerhub        Fetch repository data from Docker Hub site
     gerrit           Fetch reviews from a Gerrit server
     git              Fetch commits from Git
-    github           Fetch issues from GitHub
-    gitlab           Fetch issues from GitLab
+    github           Fetch issues, pull requests and repository information from GitHub
+    gitlab           Fetch issues, merge requests and project information from GitLab
```
Comment (valeriocos) [inline README.md]: The categories for GitLab are [issues and merge requests](https://github.com/chaoss/grimoirelab-perceval/blob/master/perceval/backends/core/gitlab.py#L87) (project info is not retrieved).

File: README.md [hunk 2]
```diff
@@ -216,6 +216,12 @@ $ perceval git '/tmp/gitlog.log'
 $ perceval github elastic logstash --from-date '2016-01-01'
 ```
 
+You can pass category as parameter which take `issue`, `pull_request` when using Github Perceval backend to get the specific data.
```
Comment (valeriocos) [inline README.md]: Category is a param available for all backends, but in this sentence it seems that that param is valid only for the GitHub backend. I would rephrase the sentence with something like:
```
The GitHub backend accepts the categories `issue`, `pull_request` and `repository` which allow to fetch the specific data.
```

what do you think @vchrombie ?

File: bin/perceval [hunk 3]
```diff
@@ -47,8 +47,8 @@ are:
     dockerhub        Fetch repository data from Docker Hub site
     gerrit           Fetch reviews from a Gerrit server
     git              Fetch commits from Git
-    github           Fetch issues from GitHub
-    gitlab           Fetch issues from GitLab
+    github           Fetch issues, pull requests and repository information from GitHub
+    gitlab           Fetch issues, merge requests and project information from GitLab
```
Comment (valeriocos) [inline bin/perceval]: Same comment as above, the categories for GitLab are issues and merge requests.

File: perceval/backends/core/github.py [hunk 4]
```diff
@@ -569,7 +569,10 @@ def issue_comments(self, issue_number):
         return self.fetch_items(path, payload)
 
     def issues(self, from_date=None):
-        """Get the issues from pagination. Note that issues contain also pull requests."""
+        """Get the issues from pagination. Note that issues contain also pull requests.
+        
+        You can filter them by using the optional arguments.
```
Comment (valeriocos) [inline perceval/backends/core/github.py]: The docstring should be improved. A docstring should be composed of:
- a first line that summarizes what the method does
- a body that explains what the method does (optiona)
- a description for each param
- a description of what the method returns (optional)
- a description of the errors that the method raise (optional)

As examples, please have a look at the methods [show](https://github.com/chaoss/grimoirelab-perceval/blob/master/perceval/backends/core/git.py#L1041),  [fetch_items](https://github.com/chaoss/grimoirelab-perceval/blob/master/perceval/backends/core/git.py#L134) and [has_archiving](https://github.com/chaoss/grimoirelab-perceval/blob/master/perceval/backends/core/git.py#L167) in git.py

File: perceval/backends/core/github.py [hunk 5]
```diff
@@ -584,7 +587,11 @@ def issues(self, from_date=None):
         return self.fetch_items(path, payload)
 
     def pulls(self, from_date=None):
-        """Get the pull requests from pagination"""
+        """Get the pull requests from pagination. Note that the pull requests are
```
Comment (valeriocos) [inline perceval/backends/core/github.py]: Same comment as above

**Comment by vchrombie** [comment]
Thanks, @jgbarah and @valeriocos. I am a bit busy these days with my University exams. I will update the changes within a couple of days.

**Comment by vchrombie** [comment]
And I have a small doubt regarding the CI build check; why is it failing?
I didn't make any code change. It was just README and comment improvements.

Is it related to my PR change? I am just curious if I did something wrong.

**Comment by inishchith** [comment]
@vchrombie I just checked the CI Build logs.
Actually the `flake8` checks fails. You can run `flake8` locally and resolve them :)

**Comment by valeriocos** [comment]
> Thanks, @jgbarah and @valeriocos. I am a bit busy these days with my University exams. I will update the changes within a couple of days.

Perfect, thanks

File: README.md (hunk 1 above)
Comment (vchrombie) [reply README.md]: Ok, I will update the change.

File: README.md (hunk 2 above)
Comment (vchrombie) [reply README.md]: Is the category param available for all the backends?

File: perceval/backends/core/github.py (hunk 4 above)
Comment (vchrombie) [reply perceval/backends/core/github.py]: Ok, cool. I learned something which is very important I guess. 
Sorry, I didn't know this before. 

I can use them in my microtasks as well. Thanks for sharing the information. :slightly_smiling_face:

**Comment by vchrombie** [comment]
@valeriocos 
I have updated this PR with the changes you have suggested the other day. Can you check and tell me if anything more is required. I am ready to update that. :slightly_smiling_face: 

@inishchith 
Thanks for the suggestion. It worked. :v:

**Comment by vchrombie** [comment]
> [Coverage Status](https://coveralls.io/builds/22549666)
> 
> Coverage decreased (-0.0005%) to 97.019% when pulling **[3b1c729](https://github.com/chaoss/grimoirelab-perceval/commit/3b1c7299682353ed0993c1ba8a867cbe33191dab) on vchrombie:update-gp-gh** into **[7dfb8b6](https://github.com/chaoss/grimoirelab-perceval/commit/7dfb8b6ddc7a85aab2347b08e5859d662d2e9b7f) on chaoss:master**.

Oh, is this a problem? Can I do anything to avoid this?

**Comment by vchrombie** [comment]
@valeriocos 
I have squashed the commits too, to maintain the readability of the commits. Please have a look and inform me if I have to change anything more.

**Comment by valeriocos** [comment]
Thank you @vchrombie for your effort. Overall it looks good to me, I would change the commit message and description with something like:
```
[doc] Update documentation for github and gitlab backends

This code updates the information about the categories of the github and gitlab backends in the README and bin/perceval. Furthermore, it improves the docstrings of the methods `pulls` and `issues` of the GitHubClient.

Signed-off-by: Venu Vardhan Reddy Tekula <venuvardhanreddytekula8@gmail.com>
```

**Comment by vchrombie** [comment]
Thanks for the suggestions. :slightly_smiling_face: 
It is good to have such meaningful commit messages and PR description. :+1:

**Comment by vchrombie** [comment]
@valeriocos 
I have updated the changes as per your suggestions.
Please have a look when you are free. :slightly_smiling_face:

**Comment by valeriocos** [comment]
Sure @vchrombie ! I see that the commit message hasn't changed:
https://github.com/chaoss/grimoirelab-perceval/pull/502/commits/1e94aa528a26e2e66ac079911d27fea1fe8ac7fd

Could you align it with the comment at: https://github.com/chaoss/grimoirelab-perceval/pull/502#issuecomment-479422547 ?

You can change the commit message in the following ways:
```
- git commit --amend (to modify the last commit)
- git push <your-branch> --force (to update your PR with the new changes)
or
- git rebase -i HEAD~1 (to modify the last commit)
- in the new editor, please change pick for r and then ctrl+O, ctrl+X (to save the changes
- in the new editor, change the commit message and then crtl+O, ctrl+X (to save the changes)
- git push <your-branch> --force (to update your PR with the new changes)
```

Thanks, let me know if you have any problem

**Comment by vchrombie** [comment]
Thanks, @valeriocos for the suggestions.

Ya, I know the procedure but I thought the second line was the PR description. Sorry, I have mistaken.
But I have updated it with the changes. You can have a look at it. :slightly_smiling_face:

**Comment by valeriocos** [comment]
Perfect @vchrombie, thank you for your time, we are almost done.

The commit message for https://github.com/chaoss/grimoirelab-perceval/pull/502/commits/d78e8300a0b6419ef3fd92449560bc7d0a0e9680 contains not needed info, the final message should be the following (where possible take care of not exceeding 72 characters per line):
```
[doc] Update documentation for github and gitlab backends

This code updates the information about the categories of the github 
and gitlab backends in the README and bin/perceval. Furthermore, it improves 
the docstrings of the methods `pulls` and `issues` of the GitHubClient.

Signed-off-by: Venu Vardhan Reddy Tekula <venuvardhanreddytekula8@gmail.com>
```

For the next time, please have a look at this blog https://chris.beams.io/posts/git-commit/ (or other similar blogs) to prepare your commit message.

**Comment by vchrombie** [comment]
Thanks, @valeriocos.

> The commit message for [d78e830](https://github.com/chaoss/grimoirelab-perceval/commit/d78e8300a0b6419ef3fd92449560bc7d0a0e9680) contains not needed info, the final message should be the following (where possible take care of not exceeding 72 characters per line):

Okay, I didn't know this thing.

> For the next PRs, please have a look at this blog https://chris.beams.io/posts/git-commit/ (or other similar blogs) to prepare your commit message :)

Thanks for the reference. I will go through it and make sure it will be implemented from next time.
I updated the changes. :slightly_smiling_face:

**Comment by valeriocos** [comment]
I still see the commit message for https://github.com/chaoss/grimoirelab-perceval/pull/502/commits/422f4b93ed7d55cd50764d1fbb3a8f7464796eb6 with not needed info. Correct if I'm wrong @vchrombie

**Comment by vchrombie** [comment]
@valeriocos 
I am so sorry. I have squashed the commits for the good readability but I totally forgot to remove that particular commit message. I overlooked it. 

Sorry about that. 
I changed it. Can you look if it is good now?

**Comment by valeriocos** [comment]
No worries @vchrombie , now it looks ok, thanks!
@sduenas if you agree, the PR could be merged.

**Comment by vchrombie** [comment]
Thanks, @valeriocos and sorry for all that mess. :sweat_smile:

**Comment by sduenas** [comment]
Thanks for your contribution. I've just rebased the changes into your branch to merge this PR.

**Comment by vchrombie** [comment]
Thank You @sduenas. :slightly_smiling_face:
